### PR TITLE
Make client methods optional

### PR DESCRIPTION
**Pull request by danieljanes**

### CODE REVIEW

File: src/py/flwr/client/app.py [hunk 1]
```diff
@@ -181,5 +220,103 @@ class `flwr.client.NumPyClient`.
...
+        member_dict["get_parameters"] = _get_parameters
+
+    if numpyclient_has_fit(client=client):
+        member_dict["fit"] = _fit
+
+    if numpyclient_has_evaluate(client=client):
+        member_dict["evaluate"] = _evaluate
+
+    # Create wrapper class with a random class name
+    wrapper_class_name = "NumPyClientWrapper" + uuid4().hex[:8]
```
Comment (tanertopal) [inline src/py/flwr/client/app.py:318]: @danieljanes why do we do this? Can you explain in the comment. I am unclear why random.

**Review by tanertopal — CHANGES_REQUESTED** [review]
Apart from one question I have it looks good to me.

File: src/py/flwr/client/app.py (hunk 1 above)
Comment (danieljanes) [reply src/py/flwr/client/app.py]: @tanertopal I looked deeper into it, it's not really necessary, so I removed it

**Review by tanertopal — APPROVED** [review]
lgtm!

**Pull request merged**